### ISSUE TITLE
MacOS install fails

### ISSUE THREAD

**Issue by Auerhahn22** [doctorfree/RoonCommandLine#6]
The installation under MacOS is impossible.  I get:

»fatal: You must specify a repository to clone.«

What can I do?

**Comment by doctorfree** [comment]
I am unable to reproduce this error on my macOS system. Can you provide additional details? Which version of macOS are you running? What was the installation process? Is your Roon Core on the same local network as your Mac?

The installation process on macOS is as follows:

```bash
git clone https://github.com/doctorfree/RoonCommandLine.git
cd RoonCommandLine
./Install
```

**Comment by Auerhahn22** [comment]
The problem was that here the command is not
git clone https://github.com/doctorfree/RoonCommandLine.git

but
git clone `https://github.com/doctorfree/RoonCommandLine.git`

When I remove the it works, and the install script begins. This works until the point where I shall activate the extension. I can’t because John the extensions page there is nothing to activate. This seems to be the end.

Werner Hintze
Am 28. Juni 2024, 19:45 +0300 schrieb Ronald Record ***@***.***>:

**Comment by Auerhahn22** [comment]
I attach a screenshot. I don’t see where I can hit ”Activate“ or something like this.

Werner Hintze
Am 28. Juni 2024, 19:45 +0300 schrieb Ronald Record ***@***.***>:

**Comment by doctorfree** [comment]
Ok, we're past the first problem. Now to activate the extension, let's see what is happening.

The `./Install` process should present you with the following lines:

```
Waiting for authorization - in Roon, click Settings -> Extensions -> Enable

2024-06-28 09:39:06,535 INFO   roonapi -- The application should be approved within Roon's settings.
```

Do you see something like that? At that point it is waiting for you to enable the extension. To do this you go to your Roon app and click the settings wheel in the top left of the Roon app window. Scroll down the settings and click `Extensions`. What do you see in the Roon application Extensions settings?

**Comment by Auerhahn22** [comment]
I See this message. That’s why I go to this page. But there is nothing. The page is just empty as you see on the screenshot I sent you.

Werner Hintze
Am 28. Juni 2024, 20:08 +0300 schrieb Ronald Record ***@***.***>:

**Comment by doctorfree** [comment]
Perhaps it is a networking issue.  Can you provide additional details? Which version of macOS are you running? Is your Roon Core on the same local network as your Mac?

Locate the IP address of your Roon Core by going to the Roon App and clicking the wheel settings icon in the upper left of the window then `General`. This should show the IP address of the Roon Core. If not, go to Settings -> Setup -> Find Roon OS and click the `Find` button.

When you discover the IP address of your Roon Core try to ping it from the system on which you are installing RoonCommandLine. For example, if the IP address of your Roon Core is `10.0.1.81` then open a terminal on your Mac and run the command:

```bash
ping 10.0.1.81
```

Substitute your Core IP address for `10.0.1.81` above.

If you are not able to ping your Core from your Mac then it is a networking issue.

If you are able to ping the Core from your Mac then try restarting your Core and re-installing RoonCommandLine.

**Comment by Auerhahn22** [comment]
I have MacOS 14.4.1 (Intel). The Roon Core is in the same network (I even didn’t know that I can have it in another. But anyway: Here is only one.)

This is true: I restarted Roon one my Mac, but not t he Roon Core on my Ubuntu-machine. O’ll try this- May be this helps.

By the way: The installation under Linux was easy. But I need the command line much more on my Mac.

Werner Hintze
Am 28. Juni 2024, 22:07 +0300 schrieb Ronald Record ***@***.***>:

**Comment by Auerhahn22** [comment]
OK. I restarted the Core. It’s the same. It simply doesn’t work.

Werner Hintze
Am 28. Juni 2024, 22:35 +0300 schrieb Werner Hintze ***@***.***>:

**Comment by doctorfree** [comment]
I'm running out of ideas. I can successfully install and activate `RoonCommandLine` on both Apple Silicon (Sonoma 14.5) and a Mac Pro Intel (Monterey 12.7.4). I don't have your exact version to test on but I don't see why that would be a problem.

Can you perform one more check for me? Run the following commands and let me know what Python modules and versions are installed in the Roon Python virtual environment:

```bash
source /usr/local/Roon/venv/bin/activate
python3 -m pip list
```

This should produce a list of modules and their versions. Could you post that here? I will continue to think about this.

**Comment by Auerhahn22** [comment]
This is the result:

Package            Version
------------------ --------
certifi            2024.6.2
charset-normalizer 3.3.2
click              8.1.7
commonmark         0.9.1
docutils           0.21.2
idna               3.7
ifaddr             0.2.0
pip                22.2.1
Pygments           2.18.0
requests           2.32.3
rich               12.6.0
rich-cli           1.8.0
rich-rst           1.3.1
roonapi            0.1.6
setuptools         63.2.0
six                1.16.0
textual            0.1.18
urllib3            2.2.2
websocket-client   1.8.0

Does this help?


Werner Hintze
Am 28. Juni 2024, 23:37 +0300 schrieb Ronald Record ***@***.***>:

**Comment by Auerhahn22** [comment]
I just tried the command get_core-Ip on the  Linux machine. It asks me to activate the extension, and I found the switch on the responding page. I activated it and it worked. But this helps nothing, I can’t install the command line on my Mac.

But this is not big problem. I didn’t want to do with it, I thought just I can build with this command line and crontab a sort of alarm clock which is the only thing I miss since I left the Sonos-app. But I see that there is no fade in option, so this idea died away. So all is well, may it work or not, I don’t need it ;)

Best regards

Werner Hintze
Am 28. Juni 2024, 23:37 +0300 schrieb Ronald Record ***@***.***>:
